components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/TokenValidationHandler.java

```diff
@@ -458,7 +460,23 @@ private OAuth2IntrospectionResponseDTO validateAccessToken(OAuth2TokenValidation
                         = OAuthServerConfiguration.getInstance().isCrossTenantTokenIntrospectionAllowed();
                 if (!isCrossTenantTokenIntrospectionAllowed && accessTokenDO != null &&
                         !tenantDomain.equalsIgnoreCase(accessTokenDO.getAuthzUser().getTenantDomain())) {
-                    throw new IllegalArgumentException("Invalid Access Token. ACTIVE access token is not found.");
+                    /*
+                    If isOrganizationManagementEnable is true, validate whether client's tenant relationship with
+                    authorized user tenant.
+                     */
+                    if (!OAuth2ServiceComponentHolder.getInstance().isOrganizationManagementEnabled()) {
+                        throw new IllegalArgumentException("Invalid Access Token. ACTIVE access token is not found.");
```

Comment:
Wondering about this exception type and error message, don't we have any better alternatives for the exception & for the error msg may be we can simply say `Invalid Access token`, WDYT?

Reply:
`IllegalArgumentException` is the exception used throughout introspection failures in this class. 
Therefore, shall we stick to the same exception?
Regarding the error message: I'm using the same message we had initially check line no: 461.

Reply:
I can see that but IMO I don't feel those are correct as well. Regarding the error msg  `ACTIVE access token is not found` this part seems a bit misleading in this particular context, ie  the token is active but they not related to this tenant that's why its okay to say the token is invalid but not the second half which states that provided token is not active